sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -97,7 +97,30 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
       options = Map.empty)
   }
 
-  protected def getExternalTmpPath(
+  /*
+   * Mostly copied from Context.java#getMRTmpPath of Hive 2.3
```

Comment:
Is this method derived from around [here](https://github.com/apache/hive/blob/branch-2.3/ql/src/java/org/apache/hadoop/hive/ql/Context.java#L446)? Or, from another place?

Reply:
Yes, it is coming from this. https://github.com/apache/hive/blob/rel/release-2.3.0/ql/src/java/org/apache/hadoop/hive/ql/Context.java#L588

Reply:
Why v2.3 instead of v3.x? Any reason?

Reply:
I believe that currently Spark's default Hive version is 2.3, so I just used the same version of Hive here for simplicity.

Reply:
I believe that currently Spark's default Hive version is 2.3, so I just used the same version of Hive here for simplicity.

Reply:
Ah, I see. IMO newer code is better. Is the `getMRTempPath` code different between 2.3 and 3.x?

Reply:
I checked `hive/Context.java` in the master branch, and I did not see much difference.
- master: https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/Context.java
- release 2.3.0: https://github.com/apache/hive/blob/rel/release-2.3.0/ql/src/java/org/apache/hadoop/hive/ql/Context.java#L588